mitodl/ocw-studio
# Google Drive sync status tracking - backend

#### Pre-Flight checklist


#### What are the relevant tickets?
#717 

#### What's this PR do?
Tracks progress of the `import_website_files` task for a site.

#### How should this be manually tested?
- Use the same values for `AWS_`, `DRIVE_` .env settings as RC
- Add some new files to a site's `files_final` subfolders in google drive.
- Click the `Sync w/Google Drive` button for the site.
- After the task runs, your site should have `sync_status='Complete'`, `sync_errors=[]`, and a populated `synced_on` date.  The new `gdrive_sync.models.DriveFile` objects should have a status of `Complete` and null `sync_error`.
- You can try raising exceptions in various places (invalid `DRIVE_SERVICE_ACCOUNT_CREDS` or `DRIVE_SHARED_ID` values, temporary `raise` lines in the modified functions) and verify that the `Website.sync_status` value is `Errored` or `Failed`, the `website.sync_errors` has a list of 1+ errors, and any created `DriveFile` objects have a status of `Failed` and a non-null `sync_error`.
- If you return the code and settings back to normal, then run again, any existing errors should be removed and the statuses updated appropriately.


#### Where should the reviewer start?
The `Website.synced_on` value will be used to display the last sync datetime on the frontend in a future PR, but it is not used for anything here.
